### PR TITLE
Rename shared lib gp_orca into orca

### PR DESCRIPTION
**Pull request by whitehawk**
Rename shared lib gp_orca into orca

This patch renames gp_orca shared lib into orca, as we'd like to avoid using
any 'gp' prefix now. Also it updates file names and internal namings to avoid
the prefix.

### CODE REVIEW

File: gpcontrib/orca/Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
-MODULE_big = gp_orca
+MODULE_big = orca
 
-OBJS = gporca.o \
+OBJS = orca.o \
 	funcs.o \
 
 PG_CPPFLAGS = \
```
Comment (hilltracer) [inline gpcontrib/orca/Makefile:6]: Most of part Orca files in project still have `gporca` naming (e.g. `I$(top_srcdir)/src/backend/gporca` is below). Is it fine? Why do we need rename only extension, without renaming most of the code?

**Review by hilltracer — COMMENTED** [review]
When I run regression tests using
```sh
docker build -t gpdb7_c:6764_after -f arenadata/Dockerfile .

docker run --name gpdb7_opt_on --rm -it -e TEST_OS=centos \
  -e MAKE_TEST_COMMAND="-k PGOPTIONS='-c optimizer=on' installcheck-world" \
  --sysctl "kernel.sem=500 1024000 200 4096" gpdb7_c:6764_after \
  /home/gpadmin/gpdb_src/concourse/scripts/ic_gpdb.bash
```
I see that some of them do not pass (see spoilers)
<details>
  <summary>Before patch</summary>
  
```diff
make: Target 'installcheck-world' not remade because of errors.

======================================================================
DIFF FILE: ../gpdb_src/contrib/auto_explain/regression.diffs
----------------------------------------------------------------------

diff -I HINT: -I CONTEXT: -I GP_IGNORE: -U3 /home/gpadmin/gpdb_src/contrib/auto_explain/expected/auto_explain_optimizer.out /home/gpadmin/gpdb_src/contrib/auto_explain/results/auto_explain.out
--- /home/gpadmin/gpdb_src/contrib/auto_explain/expected/auto_explain_optimizer.out	2024-12-02 09:50:52.894196907 +0000
+++ /home/gpadmin/gpdb_src/contrib/auto_explain/results/auto_explain.out	2024-12-02 09:50:52.895196911 +0000
@@ -103,7 +103,7 @@
                           ->  Seq Scan on auto_explain_test.t1  
                     ->  Seq Scan on auto_explain_test.t2  
 Optimizer: GPORCA
-Settings: enable_nestloop = 'on'
+Settings: enable_nestloop = 'on', optimizer = 'on'
   (slice0)    Executor Memory: 
   (slice1)    Executor Memory: 
   (slice2)    Executor Memory: 


======================================================================
DIFF FILE: ../gpdb_src/src/test/regress/regression.diffs
----------------------------------------------------------------------

diff -I HINT: -I CONTEXT: -I GP_IGNORE: -U3 /home/gpadmin/gpdb_src/src/test/regress/expected/explain_format_optimizer.out /home/gpadmin/gpdb_src/src/test/regress/results/explain_format.out
--- /home/gpadmin/gpdb_src/src/test/regress/expected/explain_format_optimizer.out	2024-12-02 08:11:29.746993727 +0000
+++ /home/gpadmin/gpdb_src/src/test/regress/results/explain_format.out	2024-12-02 08:11:29.773993244 +0000
@@ -343,6 +343,7 @@
   Optimizer: "GPORCA"
   Settings: 
     cpu_index_tuple_cost: "N.N"
+    optimizer: "on"
     random_page_cost: "N"
 (1 row)
 select explain_filter('EXPLAIN (FORMAT YAML, VERBOSE, SETTINGS ON) SELECT * from boxes;');
@@ -385,6 +386,7 @@
   Settings: 
     cpu_index_tuple_cost: "N.N"
     random_page_cost: "N"
+    optimizer: "on"
 (1 row)
 --- Check Explain Analyze YAML output that include the slices information
 select explain_filter('EXPLAIN (ANALYZE, FORMAT YAML) SELECT * from boxes LEFT JOIN apples ON apples.id = boxes.apple_id LEFT JOIN box_locations ON box_locations.id = boxes.location_id;');
@@ -576,7 +578,7 @@
         ->  Seq Scan on public.boxes  (cost=N.N..N.N rows=N width=N) (actual time=N.N..N.N rows=N loops=N)
               Output: id, apple_id, location_id
 Optimizer: GPORCA
-Settings: cpu_index_tuple_cost = 'N.N', random_page_cost = 'N'
+Settings: cpu_index_tuple_cost = 'N.N', optimizer = 'on', random_page_cost = 'N'
 Planning Time: N.N ms
   (sliceN)    Executor memory: NK bytes.
   (sliceN)    Executor memory: NK bytes avg x N workers, NK bytes max (segN).  Work_mem: NK bytes max.

real	131m2.784s
user	126m35.906s
sys	60m33.357s
```
</details>

<details>
<summary>  After patch </summary>

```diff
make: Target 'installcheck-world' not remade because of errors.

======================================================================
DIFF FILE: ../gpdb_src/contrib/auto_explain/regression.diffs
----------------------------------------------------------------------

diff -I HINT: -I CONTEXT: -I GP_IGNORE: -U3 /home/gpadmin/gpdb_src/contrib/auto_explain/expected/auto_explain_optimizer.out /home/gpadmin/gpdb_src/contrib/auto_explain/results/auto_explain.out
--- /home/gpadmin/gpdb_src/contrib/auto_explain/expected/auto_explain_optimizer.out	2024-12-02 12:33:01.393037742 +0000
+++ /home/gpadmin/gpdb_src/contrib/auto_explain/results/auto_explain.out	2024-12-02 12:33:01.395037717 +0000
@@ -103,7 +103,7 @@
                           ->  Seq Scan on auto_explain_test.t1  
                     ->  Seq Scan on auto_explain_test.t2  
 Optimizer: GPORCA
-Settings: enable_nestloop = 'on'
+Settings: enable_nestloop = 'on', optimizer = 'on'
   (slice0)    Executor Memory: 
   (slice1)    Executor Memory: 
   (slice2)    Executor Memory: 


======================================================================
DIFF FILE: ../gpdb_src/src/test/regress/regression.diffs
----------------------------------------------------------------------

diff -I HINT: -I CONTEXT: -I GP_IGNORE: -U3 /home/gpadmin/gpdb_src/src/test/regress/expected/explain_format_optimizer.out /home/gpadmin/gpdb_src/src/test/regress/results/explain_format.out
--- /home/gpadmin/gpdb_src/src/test/regress/expected/explain_format_optimizer.out	2024-12-02 10:43:32.565617940 +0000
+++ /home/gpadmin/gpdb_src/src/test/regress/results/explain_format.out	2024-12-02 10:43:32.590629133 +0000
@@ -343,6 +343,7 @@
   Optimizer: "GPORCA"
   Settings: 
     cpu_index_tuple_cost: "N.N"
+    optimizer: "on"
     random_page_cost: "N"
 (1 row)
 select explain_filter('EXPLAIN (FORMAT YAML, VERBOSE, SETTINGS ON) SELECT * from boxes;');
@@ -385,6 +386,7 @@
   Settings: 
     cpu_index_tuple_cost: "N.N"
     random_page_cost: "N"
+    optimizer: "on"
 (1 row)
 --- Check Explain Analyze YAML output that include the slices information
 select explain_filter('EXPLAIN (ANALYZE, FORMAT YAML) SELECT * from boxes LEFT JOIN apples ON apples.id = boxes.apple_id LEFT JOIN box_locations ON box_locations.id = boxes.location_id;');
@@ -576,7 +578,7 @@
         ->  Seq Scan on public.boxes  (cost=N.N..N.N rows=N width=N) (actual time=N.N..N.N rows=N loops=N)
               Output: id, apple_id, location_id
 Optimizer: GPORCA
-Settings: cpu_index_tuple_cost = 'N.N', random_page_cost = 'N'
+Settings: cpu_index_tuple_cost = 'N.N', optimizer = 'on', random_page_cost = 'N'
 Planning Time: N.N ms
   (sliceN)    Executor memory: NK bytes.
   (sliceN)    Executor memory: NK bytes avg x N workers, NK bytes max (segN).  Work_mem: NK bytes max.
diff -I HINT: -I CONTEXT: -I GP_IGNORE: -U3 /home/gpadmin/gpdb_src/src/test/regress/expected/resource_queue.out /home/gpadmin/gpdb_src/src/test/regress/results/resource_queue.out
--- /home/gpadmin/gpdb_src/src/test/regress/expected/resource_queue.out	2024-12-02 11:20:00.865434330 +0000
+++ /home/gpadmin/gpdb_src/src/test/regress/results/resource_queue.out	2024-12-02 11:20:00.877434255 +0000
@@ -645,7 +646,5 @@
 -- run query using non_superuser role so that it can be
 -- controled by resource queue
 \! ./extended_protocol_resqueue dbname=regression extend_protocol_requeue_role;
-NOTICE:  SPI memory reservation 
-NOTICE:  query requested XXXKB
-NOTICE:  query requested XXXKB of memory
+./extended_protocol_resqueue: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by ./extended_protocol_resqueue)
 drop role extend_protocol_requeue_role;


======================================================================
DIFF FILE: ../gpdb_src/src/test/isolation2/regression.diffs
----------------------------------------------------------------------

diff -I HINT: -I CONTEXT: -I GP_IGNORE: -U3 /home/gpadmin/gpdb_src/src/test/isolation2/expected/uao/ao_unique_index_vacuum_column.out /home/gpadmin/gpdb_src/src/test/isolation2/results/uao/ao_unique_index_vacuum_column.out
--- /home/gpadmin/gpdb_src/src/test/isolation2/expected/uao/ao_unique_index_vacuum_column.out	2024-12-02 12:28:03.596476420 +0000
+++ /home/gpadmin/gpdb_src/src/test/isolation2/results/uao/ao_unique_index_vacuum_column.out	2024-12-02 12:28:03.599476344 +0000
@@ -96,12 +96,8 @@
 
 -- Wait until all old index entries have been deleted by the VACUUM.
 SELECT gp_wait_until_triggered_fault('vacuum_ao_after_index_delete', 1, dbid) FROM gp_segment_configuration WHERE role = 'p' AND content <> -1;
- gp_wait_until_triggered_fault 
-------------
- Success:                      
- Success:                      
- Success:                      
-(3 rows)
+DETAIL:  Timed-out as 10 minutes max wait happens until triggered. (gp_inject_fault.c:157)
+ERROR:  failed to inject fault: ERROR:  fault not triggered, fault name:'vacuum_ao_after_index_delete' fault type:'wait_until_triggered'
 
 -- Now trying to insert key = 1 will also be detected as a conflict, even
 -- though the old index entries are no longer present. We have the new index


real	367m42.328s
user	121m29.628s
sys	56m43.127s
```
</details>

**Is it expected behavior?**

P.S. unit_tests_gporca.bash works fine
```sh
docker run --rm -it gpdb7_c:6764_after bash -c "gpdb_src/concourse/scripts/unit_tests_gporca.bash"
...
100% tests passed, 0 tests failed out of 179
Total Test time (real) = 538.62 sec
```

**Comment by whitehawk** [comment]
> I see that some of them do not pass

1. for tests `explain_format` and `auto_explain` there is already running a review (in scope of [ADBDEV-6792](https://tracker.yandex.ru/ADBDEV-6792))
2. your result in `resource_queue` I couldn't reproduce on my side. And it doesn't look related to the renaming done in this PR. So I guess it is some flaky result.
3. fail in test `ao_unique_index_vacuum_column` I think is the same as flaky [ADBDEV-5324](https://tracker.yandex.ru/ADBDEV-5324)

File: gpcontrib/orca/Makefile (hunk 1 above)
Comment (whitehawk) [reply gpcontrib/orca/Makefile:6]: Yes, it is fine. I'm aiming to rename only newly added code. I do not want to rename the old Orca code in scope of moving it into a separate lib. I just want not to introduce new code with old naming.

**Review by hilltracer — APPROVED** [review]

**Review by bimboterminator1 — APPROVED** [review]

**Pull request merged**